lib/modules/versioning/debian/index.ts

```diff
@@ -40,31 +40,7 @@ export class DebianVersioningApi extends GenericVersioningApi {
     return this._distroInfo.isReleased(ver) && !this._distroInfo.isEolLts(ver);
   }
 
-  override getNewValue({
-    currentValue,
-    rangeStrategy,
-    newVersion,
-  }: NewValueConfig): string {
-    if (rangeStrategy === 'pin') {
-      let newVer = newVersion;
-
-      // convert newVersion to semVer
-      if (this._distroInfo.isCodename(newVersion)) {
-        newVer = this._distroInfo.getVersionByCodename(newVersion);
-      }
-      if (this._rollingReleases.has(newVersion)) {
-        newVer = this._rollingReleases.getVersionByLts(newVersion);
-      }
-
-      // current value is codename or [oldold|old|]stable
-      if (
-        this._distroInfo.isCodename(currentValue) ||
-        this._rollingReleases.has(currentValue)
-      ) {
-        return newVer;
-      }
-    }
```

Comment:
I think we lose to pin a codename to number, as it happens now with this range strategy.

Did you tested it on real repo?

Reply:
Yes, let's test it, because I think it should work automatically. e.g. currentValue will be a codename, currentVersion will be a number, so pinning will mean it's replaced with a number